File: techflix/server-wsl.py
#!/usr/bin/env python3
import http.server
import socketserver
import os
import subprocess
import socket

PORT = 8081
DIRECTORY = "dist"

class CORSRequestHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)
    
    def end_headers(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.send_header('Cache-Control', 'no-cache, no-store, must-revalidate')
        super().end_headers()

    def do_OPTIONS(self):
        self.send_response(200)
        self.end_headers()

Handler = CORSRequestHandler

# Get WSL IP
try:
    wsl_ip = subprocess.check_output(['hostname', '-I']).decode().strip().split()[0]
except:
    wsl_ip = "WSL IP not found"

# Get Windows host IP (usually the default gateway in WSL2)
try:
    windows_ip = subprocess.check_output(['ip', 'route', 'show']).decode()
    windows_ip = windows_ip.split('default via ')[1].split()[0]
except:
    windows_ip = "Windows IP not found"

print("\n🚀 TechFlix Server Starting...\n")
print(f"Server will run on port {PORT}")
print(f"Serving directory: {os.path.abspath(DIRECTORY)}\n")
print("=" * 50)
print("Access TechFlix from Windows browser at:")
print(f"  http://localhost:{PORT}")
print(f"  http://127.0.0.1:{PORT}")
print(f"  http://{wsl_ip}:{PORT}")
print("\nIf localhost doesn't work in Windows:")
print("1. Try the WSL IP address directly")
print("2. Check Windows Firewall settings")
print("3. Run this PowerShell command as Admin:")
print(f"   netsh interface portproxy add v4tov4 listenport={PORT} listenaddress=0.0.0.0 connectport={PORT} connectaddress={wsl_ip}")
print("=" * 50)
print("\nPress Ctrl+C to stop the server\n")

with socketserver.TCPServer(("0.0.0.0", PORT), Handler) as httpd:
    httpd.serve_forever()

File: techflix/server-wsl.js
const express = require('express');
const path = require('path');
const { exec } = require('child_process');

const app = express();
const PORT = 8081;

// Serve static files from dist directory
app.use(express.static(path.join(__dirname, 'dist')));

// Handle client-side routing
app.get('*', (req, res) => {
  res.sendFile(path.join(__dirname, 'dist', 'index.html'));
});

// Get WSL IP
exec('hostname -I', (error, stdout) => {
  const wslIp = stdout.trim().split(' ')[0];
  
  app.listen(PORT, '0.0.0.0', () => {
    console.log('\n🚀 TechFlix Server Running!\n');
    console.log('Access from Windows browser at:');
    console.log(`  http://localhost:${PORT}`);
    console.log(`  http://127.0.0.1:${PORT}`);
    console.log(`  http://${wslIp}:${PORT}`);
    console.log('\nIf localhost doesn\'t work, try:');
    console.log('1. Use the WSL IP address directly');
    console.log('2. Run the PowerShell script (run-wsl.ps1) as Administrator on Windows');
    console.log('3. Or use: npx localtunnel --port 8081\n');
    console.log('Press Ctrl+C to stop the server\n');
  });
});